In [188]:
import pandas as pd
import numpy as np
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline 

**1. Read the CSV file in Pandas and create a DataFrame named Grc_df. What is the number of
rows and columns in Grc_df? Print the first 10 and last 10 rows of Grc_df.**

In [189]:
# Read CSV file and create a DataFrame called Grc_df
Grc_df = pd.read_csv('Grocery_dataset.csv')

# Find number of rows and columns
Grc_df.shape

(5000, 12)

Grc_df has 5000 rows and 12 columns

In [190]:
# Print the first 10 rows
Grc_df.head(10)

,Item_Identifier,Item_Weight,Item_Fat_Content,Item_Visibility,Item_Type,Item_MRP,Outlet_Identifier,Outlet_Establishment_Year,Outlet_Size,Outlet_Location_Type,Outlet_Type,Item_Outlet_Sales
0,FDA15,9.300,Low Fat,0.016047,Dairy,249.8092,OUT049,1999,Medium,Tier 1,Supermarket Type1,3735.1380
1,DRC01,5.920,Regular,0.019278,Soft Drinks,48.2692,OUT018,2009,Medium,Tier 3,Supermarket Type2,443.4228
2,FDN15,17.500,Low Fat,0.016760,Meat,141.6180,OUT049,1999,Medium,Tier 1,Supermarket Type1,2097.2700
3,FDX07,19.200,Regular,0.000000,Fruits and Vegetables,182.0950,OUT010,1998,NaN,Tier 3,Grocery Store,732.3800
4,NCD19,8.930,Low Fat,0.000000,Household,53.8614,OUT013,1987,High,Tier 3,Supermarket Type1,994.7052
5,FDP36,10.395,Regular,0.000000,Baking Goods,51.4008,OUT018,2009,Medium,Tier 3,Supermarket Type2,556.6088
6,FDO10,13.650,Regular,0.012741,Snack Foods,57.6588,OUT013,1987,High,Tier 3,Supermarket Type1,343.5528
7,FDP10,NaN,Low Fat,0.127470,Snack Foods,107.7622,OUT027,1985,Medium,Tier 3,Supermarket Type3,4022.7636
8,FDH17,16.200,Regular,0.016687,Frozen Foods,96.9726,OUT045,2002,NaN,Tier 2,Supermarket Type1,1076.5986
9,FDU28,19.200,Regular,0.094450,Frozen Foods,187.8214,OUT017,2007,NaN,Tier 2,Supermarket Type1,4710.5350


In [191]:
# Print the last 10 rows
Grc_df.tail(10)

,Item_Identifier,Item_Weight,Item_Fat_Content,Item_Visibility,Item_Type,Item_MRP,Outlet_Identifier,Outlet_Establishment_Year,Outlet_Size,Outlet_Location_Type,Outlet_Type,Item_Outlet_Sales
4990,FDE46,18.60,Low Fat,0.015859,Snack Foods,152.9366,OUT017,2007,NaN,Tier 2,Supermarket Type1,453.4098
4991,NCQ53,17.60,Low Fat,0.018905,Health and Hygiene,234.6590,OUT046,1997,Small,Tier 1,Supermarket Type1,8508.9240
4992,NCN42,20.25,Low Fat,0.014281,Household,148.0418,OUT018,2009,Medium,Tier 3,Supermarket Type2,1177.1344
4993,DRH11,5.98,Low Fat,0.075675,Hard Drinks,53.3614,OUT049,1999,Medium,Tier 1,Supermarket Type1,331.5684
4994,NCI42,18.75,Low Fat,0.010382,Household,207.8954,OUT049,1999,Medium,Tier 1,Supermarket Type1,2292.3494
4995,FDT07,5.82,Regular,0.077476,Fruits and Vegetables,255.3330,OUT045,2002,NaN,Tier 2,Supermarket Type1,4870.3270
4996,DRL60,8.52,Low Fat,0.027059,Soft Drinks,153.3682,OUT046,1997,Small,Tier 1,Supermarket Type1,914.8092
4997,FDG20,15.50,Regular,0.126200,Fruits and Vegetables,178.4028,OUT018,2009,Medium,Tier 3,Supermarket Type2,1239.7196
4998,FDF56,16.70,Regular,0.119462,Fruits and Vegetables,182.7976,OUT046,1997,Small,Tier 1,Supermarket Type1,1810.9760
4999,FDV33,9.60,Regular,0.027455,Snack Foods,258.1304,OUT018,2009,Medium,Tier 3,Supermarket Type2,2324.9736


**2. Are there any null values in the Grc_df? If yes, then in which columns and how many?
Finally, handle these null values using any strategy shown during the labs.**

In [192]:
# Find number of null values
Grc_df.isna().sum()

Item_Identifier                 0
Item_Weight                   818
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  1439
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

Column Item_Weight has 818 null values and column Outlet_Size has 1439 null values

In [193]:
# Handle null values by dropping all rows that contain them
Grc_df = Grc_df.dropna()
# Reset index to tidy up
Grc_df = Grc_df.reset_index(drop=True)
# Check no more null values remain
Grc_df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

**3. How many unique Outlet Sizes are there in the Grc_df? Which outlet size is maximum, and
which is minimum?** 


In [194]:
# Find number of unique Outlet Sizes
Grc_df.Outlet_Size.unique()

<StringArray>
['Medium', 'High', 'Small']
Length: 3, dtype: str

There are three unique Outlet Sizes

In [195]:
# Calculate frequency of each Outlet size category
Grc_df['Outlet_Size'].value_counts()

Outlet_Size
Small     1118
Medium    1067
High       558
Name: count, dtype: int64

Not entirely what "maximum" and "minimum" would mean in this context, so found number of instances of each Outlet size category. Small has most instances with 1118 = maximum. High has least instances with 558 = minimum.

**4. How many unique Item Fat Content types are in the Grc_df? List them. Do you see any
issues with the Item Fat Content types? If yes, then handle this issue.**

In [196]:
# Find number of unique Fat Content types
Grc_df.Item_Fat_Content.unique()

<StringArray>
['Low Fat', 'Regular', 'low fat', 'reg', 'LF']
Length: 5, dtype: str

Issue is that there should only be two Fat Content types, but the Low Fat type is recorded in three different ways (Low Fat, low fat and LF) and the Regular type is recorded in two different ways (Regular, reg)

In [197]:
# Replace duplicate values
Grc_df["Item_Fat_Content"] = Grc_df["Item_Fat_Content"].replace('low fat', 'Low Fat')
Grc_df["Item_Fat_Content"] = Grc_df["Item_Fat_Content"].replace('LF', 'Low Fat')
Grc_df["Item_Fat_Content"] = Grc_df["Item_Fat_Content"].replace('reg', 'Regular')
# Check only two Fat Context types after change
Grc_df.Item_Fat_Content.unique()

<StringArray>
['Low Fat', 'Regular']
Length: 2, dtype: str

**5. Drop the columns having index values of 0, 6 and create a new DataFrame Grc_new_df.**

In [198]:
# Use .drop() to drop columns by index no.
Grc_new_df = Grc_df.drop(Grc_df.columns[[0, 6]], axis=1)

**6. Using different Supermarket type listed in the column Outlet_Type create two different
DataFrames from Grc_new_df. Name these DataFrames as SupType_1 and SupType_2.**


In [199]:
# Save two Outlet Types to array
s_types = Grc_new_df['Outlet_Type'].unique()
# Create new DataFrames using values from array
SupType_1 = Grc_new_df[Grc_new_df['Outlet_Type'] == s_types[0]]
SupType_2 = Grc_new_df[Grc_new_df['Outlet_Type'] == s_types[1]]

**7. Using Seaborn (“ggplot style”) create a (2,1) subplot of a box plot showing 5-point summary of the column Item_MRP for SupType_1 and SupType_2. Which Outlet Type has a higher median MRP? Are there any outliers?**

_Don't really understand the question_

**8. Concatenate the DataFrames SupType_1 and SupType_2 to create a new DataFrame Grc_Concat_df and sort it in ascending order based on the column Item_Outlet_Sales. What is the Outlet_Location_type of the store with the lowest sales?**

In [200]:
# Concatenate two DataFrames and create new
Grc_Concat_df = pd.concat([SupType_1, SupType_2], ignore_index=True)
# Sort in asc. order based on Item_Outlet_Sales
Grc_Concat_df = Grc_Concat_df.sort_values('Item_Outlet_Sales', ascending=True)
# Find Outlet_Location_type using index 0 as DataFrame is sorted
Grc_Concat_df.iloc[0]['Outlet_Location_Type']

'Tier 3'

The Outlet_Location_type of the store with the lowest sales is Tier 3.

**9. Using columns Outlet_Size and Outlet_Location_Type to create a multiple index for
Grc_Concat_df.**

_Question is incoherent_